Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -9748,7 +9748,39 @@ - (UITargetedPreview *)_createTargetedContextMenuHintPreviewForFocusedElement
         }
     }();
 
-    auto targetedPreview = createFallbackTargetedPreview(self, self.containerForContextMenuHintPreviews, _focusedElementInformation.interactionRect, backgroundColor);
+    auto previewRect = _focusedElementInformation.interactionRect;
+    if (positioning == WebKit::TargetedPreviewPositioning::LeadingOrTrailingEdge) {
+        static constexpr auto defaultMenuWidth = 250;
+        auto unobscuredRect = WebCore::IntRect { self.unobscuredContentRect };
+        std::optional<int> previewOffsetX;
+
+        auto leftEdge = previewRect.x() - defaultMenuWidth;
+        bool hasSpaceAfterRightEdge = previewRect.maxX() + defaultMenuWidth <= unobscuredRect.maxX();
+        bool hasSpaceBeforeLeftEdge = leftEdge > unobscuredRect.x();
+
+        switch (self.effectiveUserInterfaceLayoutDirection) {
+        case UIUserInterfaceLayoutDirectionLeftToRight:
```

Comment:
I think this should really be about the direction of the focused element and not the view.

But this was probably wrong even before this change, so should not block landing.

Reply:
Oh, good point. I filed [this bug](https://bugs.webkit.org/show_bug.cgi?id=264386) to track that.

(As you noted, the existing implementation already doesn't handle this case)